File: Examples/BcastRecCompOrdBcastWithResRec/src/course/examples/broadcastreceiver/compoundorderedbroadcast/Receiver2.java
package course.examples.broadcastreceiver.compoundorderedbroadcast;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class Receiver2 extends BroadcastReceiver {

	private final String TAG = "Receiver2";

	@Override
	public void onReceive(Context context, Intent intent) {

		Log.i(TAG, "INTENT RECEIVED by Receiver2");

		String tmp = getResultData() == null ? "" : getResultData();
		setResultData(tmp + "Receiver 2:");
	}

}

File: ExamplesKotlin/BcastRecCompOrdBcastWithResRec/app/src/main/java/course/examples/broadcastreceiver/resultreceiver/Receiver2.kt
package course.examples.broadcastreceiver.resultreceiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class Receiver2 : BroadcastReceiver() {

    companion object {
        private const val TAG = "Receiver2"
    }

    override fun onReceive(context: Context, intent: Intent) {

        Log.i(TAG, "INTENT RECEIVED by Receiver2")

        val tmp = if (resultData == null) "" else resultData
        resultData = "$tmp:Receiver 2:"
    }

}
